lib/backend/syncersv1/updateprocessors/globalnetworkpolicyprocessor.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package updateprocessors
+
+import (
+	"errors"
+
+	"github.com/projectcalico/libcalico-go/lib/apiv2"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/backend/watchersyncer"
+)
+
+// Create a new SyncerUpdateProcessor to sync GlobalNetworkPolicy data in v1 format for
+// consumption by Felix.
+func NewGlobalNetworkPolicyUpdateProcessor() watchersyncer.SyncerUpdateProcessor {
+	return NewGeneralUpdateProcessor(apiv2.KindGlobalNetworkPolicy, convertGlobalNetworkPolicyV2ToV1)
+}
+
+// Convert v2 KVPair to the equivalent v1 KVPair.
+func convertGlobalNetworkPolicyV2ToV1(kvp *model.KVPair) (*model.KVPair, error) {
+	// Validate against incorrect key/value kinds.  This indicates a code bug rather
+	// than a user error.
+	v2key, ok := kvp.Key.(model.ResourceKey)
+	if !ok || v2key.Kind != apiv2.KindGlobalNetworkPolicy {
+		return nil, errors.New("Key is not a valid GlobalNetworkPolicy resource key")
+	}
+
+	if kvp.Value == nil {
```

Comment:
I hope this isn't true, because for a delete event, the value will always be nil.

For resources where the name of the v2 resource can be mapped directly to the v1 index then the v2 key will have sufficient information to construct the v1 key.  In this case the v2Key name basically maps to the name in the v1 key.  You don't need the resource to extract that.

Reply:
Ok, I was using the Value for the name, but if it is always nil for delete, then I'll have to derive the name from the key.  Is the Key name always guaranteed to be the same as the name in the Value?

Reply:
yup it will be unless someone is hand editing stuff in the datastore.

Reply:
same with the namespace.